lib/rules/no-wait-for-side-effects.ts

```diff
@@ -26,6 +26,28 @@ export default createTestingLibraryRule<Options, MessageIds>({
   },
   defaultOptions: [],
   create: function (context, _, helpers) {
+    function isCallerWaitFor(
+      node: TSESTree.BlockStatement | TSESTree.CallExpression
+    ) {
+      if (!node.parent) {
+        return false;
+      }
+      const callExpressionNode = node.parent.parent as TSESTree.CallExpression;
+      const callExpressionIdentifier = getPropertyIdentifierNode(
+        callExpressionNode
+      );
+
+      if (!callExpressionIdentifier) {
+        return false;
+      }
+
+      if (!helpers.isAsyncUtil(callExpressionIdentifier, ['waitFor'])) {
+        return false;
+      }
+
+      return true;
```

Comment:
No need for early return anymore in this part

```suggestion
      return Boolean(callExpressionIdentifier) &&
        helpers.isAsyncUtil(callExpressionIdentifier, ['waitFor']);
```

Reply:
![image](https://user-images.githubusercontent.com/34719980/115884893-1026fd00-a458-11eb-8290-5da1527c4e35.png)

Because of  ⬆️ , I'd vote for this:

```
      return (
        !!callExpressionIdentifier &&
        helpers.isAsyncUtil(callExpressionIdentifier, ['waitFor'])
      );
```

What do you think?

Reply:
Let's use double negation for now then

Reply:
I think we can leave it as it was since 1) it works and 2) it's an inherited code, you just moved it somewhere else.

Reply:
@zaicevas oh wow, if you cast it with `Boolean` TS complains, but if you cast with `!!` TS doesn't complain? :O

Reply:
Yep 😄

Reply:
Oh wow, TIL https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAcwE4FN2zMgFGAQwFt0AuRAZylRhwEpyqadEBvAKES8QyhFSQADABLoANmLiIAJK0IkAvoPYL27MVkRg4AOWJlK1WskQAfLSAmIAvBYkBuNQCE4cDQTD5d+uogBkfigYWMZeeiR07ACEUWE+-oFomNh42uHodEA

Reply:
Added double negation 8e6044a